tests/fixture/test_data_server_test.go

```diff
@@ -34,38 +34,44 @@ func TestAllocateFundedKeys(t *testing.T) {
 		name              string
 		count             int
 		expectedAddresses []ids.ShortID
-	}{{
-		name:  "single key",
-		count: 1,
-		expectedAddresses: []ids.ShortID{
-			keys[4].Address(),
+		expectedError     error
+	}{
+		{
+			name:  "single key",
+			count: 1,
+			expectedAddresses: []ids.ShortID{
+				keys[4].Address(),
+			},
+			expectedError: nil,
 		},
-	}, {
-		name:  "multiple keys",
-		count: 4,
-		expectedAddresses: []ids.ShortID{
-			keys[0].Address(),
-			keys[1].Address(),
-			keys[2].Address(),
-			keys[3].Address(),
+		{
+			name:  "multiple keys",
+			count: 4,
+			expectedAddresses: []ids.ShortID{
+				keys[0].Address(),
+				keys[1].Address(),
+				keys[2].Address(),
+				keys[3].Address(),
+			},
+			expectedError: nil,
 		},
-	}, {
-		name:  "insufficient keys available",
-		count: 1,
-	}}
+		{
+			name:              "insufficient keys available",
+			count:             1,
+			expectedAddresses: []ids.ShortID{},
+			expectedError:     errRequestedKeyCountExceedsAvailable,
+		},
+	}
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			keys, err := AllocateFundedKeys(uri, tc.count)
-			if tc.expectedAddresses == nil {
-				require.ErrorIs(err, errRequestedKeyCountExceedsAvailable)
-			} else {
-				require.NoError(err)
-				addresses := []ids.ShortID{}
-				for _, key := range keys {
-					addresses = append(addresses, key.Address())
-				}
-				require.Equal(tc.expectedAddresses, addresses)
+			require.ErrorIs(err, tc.expectedError)
+
+			addresses := make([]ids.ShortID, len(keys))
+			for i, key := range keys {
+				addresses[i] = key.Address()
 			}
+			require.Equal(tc.expectedAddresses, addresses)
```

Comment:
This is a bit odd as it enforces that `keys` is empty if an error is returned... If we want to remove this requirement than I'd prefer adding a check after the `require.ErrorIs(err, tc.expectedError)` with `if err != nil { return }`